src/nupic/algorithms/SpatialPooler.hpp

```diff
@@ -772,9 +770,8 @@ class SpatialPooler : public Serializable
   // Implementation methods. all methods below this line are
   // NOT part of the public API
 
-  void toDense_(vector<UInt> &sparse, UInt dense[], UInt n);
 
-  void boostOverlaps_(vector<UInt> &overlaps, vector<Real> &boostedOverlaps);
+  void boostOverlaps_(const vector<UInt> &overlaps, vector<Real> &boostedOverlaps) const;
```

Comment:
making this much const seemed for better optimization, roughly 10-20% faster here